ebpf/state.go

```diff
@@ -171,64 +157,32 @@ func (ns *networkState) StoreClosedConnection(conn ConnectionStats) {
 	ns.Lock()
 	defer ns.Unlock()
 
-	now := time.Now()
 	key, err := conn.ByteKey(ns.buf)
 	if err != nil {
 		log.Warn("failed to create byte key: %s", err)
 		return
 	}
 
 	for _, client := range ns.clients {
+		// If we've seen this closed connection already, lets combine the two
 		if prev, ok := client.closedConnections[string(key)]; ok {
-			// Entry is already here, add the old connection to override connections
-			// For later aggregation
-			newOverride := statsFromConn(*prev, now)
-			if override, ok := client.overrideConnections[string(key)]; ok {
-				// If we already have an override aggregate the two overrides
-				newOverride = aggregateStats(override, newOverride, now)
+			// We received either the connections either out of order, or it's the same one we've already seen.
+			// Lets skip it for now.
+			if prev.LastUpdateEpoch >= conn.LastUpdateEpoch {
+				ns.telemetry.unorderedConns++
+				continue
 			}
-			client.overrideConnections[string(key)] = newOverride
-		}
-
-		// We only store the pointer to the connection, when it will be cleared for each client it will get GCed
-		client.closedConnections[string(key)] = &conn
-	}
-}
-
-// closedConns returns the closed connections for the given client and takes care of updating last fetch
-// the provided client is supposed to exist
-func (ns *networkState) closedConns(clientID string) []ConnectionStats {
-	client := ns.clients[clientID]
-	conns := make([]ConnectionStats, 0, len(client.closedConnections))
-
-	for key, conn := range client.closedConnections {
-		c := *conn
 
-		// First check if we have an override stored
-		// If we do aggregate it and delete the override
-		if override, ok := client.overrideConnections[key]; ok {
-			aggregateConnAndStat(&c, &override)
-			delete(client.overrideConnections, key)
-		}
-
-		if stats, ok := client.stats[key]; ok {
-			c.LastSentBytes = c.MonotonicSentBytes - stats.totalSent
-			c.LastRecvBytes = c.MonotonicRecvBytes - stats.totalRecv
-			c.LastRetransmits = c.MonotonicRetransmits - stats.totalRetransmits
-			delete(client.stats, key)
-		} else {
-			c.LastSentBytes = c.MonotonicSentBytes
-			c.LastRecvBytes = c.MonotonicRecvBytes
-			c.LastRetransmits = c.MonotonicRetransmits
+			conn.MonotonicSentBytes += prev.MonotonicSentBytes
+			conn.MonotonicRecvBytes += prev.MonotonicRecvBytes
+			conn.MonotonicRetransmits += prev.MonotonicRetransmits
+		} else if len(client.closedConnections) >= maxClientClosedConns {
+			ns.telemetry.connsDropped++
```

Comment:
Would it be worth also logging here ? 
To know if we have to increase the maxClientClosedConns since we want to make it configurable (Not every time a connection is dropped though)

Reply:
I put this in a debug regular 5m log flush (look at other references to `connsDropped`). I might change it to INFO, but will consider that later. 

Not worth logging here because we'll fill up logs.